integration/spark2/src/main/scala/org/apache/spark/sql/CarbonDataFrameWriter.scala

```diff
@@ -92,11 +89,35 @@ class CarbonDataFrameWriter(sqlContext: SQLContext, val dataFrame: DataFrame) {
     ).filter(_._2.isDefined)
       .map(property => s"'${property._1}' = '${property._2.get}'").mkString(",")
 
+    val partitionColumns: Seq[String] = if (options.partitionColumns.isDefined) {
+      options.partitionColumns.get.map { column =>
+        val c = schema.fields.find(_.name.equalsIgnoreCase(column))
+        if (c.isEmpty) {
+          throw new MalformedCarbonCommandException(s"invalid partition column: $column")
```

Comment:
missing validation for duplicated column names?

Reply:
fixed